### PR TITLE
Add metadata schema methods to Python C module

### PR DESCRIPTION
**Pull request by benjeffery**
As #491 was getting very large we decided to split it again. This PR adds methods to the Python C module to get/set metadata schema as a `*Table` attribute, and to the get the group of schemas as a method on `TreeSequence`

### CODE REVIEW

File: python/_tskitmodule.c [hunk 1]
```diff
@@ -6303,6 +6670,46 @@ TreeSequence_get_site(TreeSequence *self, PyObject *args)
     return ret;
 }
 
+static PyObject *
+TreeSequence_get_metadata_schemas(TreeSequence *self) {
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6674]: Looking ahead a bit, maybe we should call this ``get_table_metadata_schemas``? It'll be a bit confusing to have both ``get_metadata_schema`` and ``get_metadata_schemas``.

File: python/tests/test_lowlevel.py [hunk 2]
```diff
@@ -445,6 +457,30 @@ def test_mean_descendants(self):
             A = ts.mean_descendants([focal[2:], focal[:2]])
             self.assertEqual(A.shape, (ts.get_num_nodes(), 2))
 
+    def test_metadata_schemas(self):
+        tables = _tskit.TableCollection(1.0)
+        # Set the schema
+        for table_name in self.metadata_tables:
+            getattr(
```
Comment (jeromekelleher) [inline python/tests/test_lowlevel.py:464]: I'd find this more readable as 
```python
table = getattr(tables, f"{table_name}s")
table.metadata_schema = f"{table_name} test metadata schema"
```
Had to read it back a couple of times to figure out what it was doing.

File: python/tests/test_lowlevel.py [hunk 3]
```diff
@@ -2013,6 +2049,66 @@ def test_map_mutations_errors(self):
...
 
+class TestTableMetadataSchema(unittest.TestCase, MetadataTestMixin):
+    def test_metadata_schema_attribute(self):
+        tables = _tskit.TableCollection(1.0)
+        for table in self.metadata_tables:
+            table = getattr(tables, f"{table}s")
+            # Check default value
+            self.assertEqual(table.metadata_schema, "")
+            # Set and read back
+            example = "An example of metadata schema with unicode 🎄🌳🌴🌲🎋"
```
Comment (jeromekelleher) [inline python/tests/test_lowlevel.py:2060]: Just spotted that these are all different types of unicode tree. Nice!

File: python/_tskitmodule.c (hunk 1 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: Good idea, fixed in https://github.com/tskit-dev/tskit/pull/542/commits/f89d21cfcc3e12219eb3fc0819e0e16a7370beac

File: python/tests/test_lowlevel.py (hunk 2 above)
Comment (benjeffery) [reply python/tests/test_lowlevel.py]: Yeah, line breaks don't help. Fixed in https://github.com/tskit-dev/tskit/pull/542/commits/fc229a4e3687baa3ba025123add768a2597efacb

File: python/tests/test_lowlevel.py (hunk 3 above)
Comment (benjeffery) [reply python/tests/test_lowlevel.py:2059]: Tree-mendous. These little things keep me sane.

**Comment by benjeffery** [comment]
@jeromekelleher looks good to me:
```
(env) benj@treebeard:~/projects/tskit/python$ python3 stress_lowlevel.py -m lowlevel
iter	tests	err	fail	skip	RSS	min	max	max@iter
317	166	0	0	0	78668	78336	78668	2
```
If you're happy I'll squash rebase.

**Comment by jeromekelleher** [comment]
Great thanks, let's merge.

**Comment by benjeffery** [comment]
Travis failure is on `push` trigger only. `pr` trigger is passing.

**Pull request merged**